src/librustdoc/clean/blanket_impl.rs

```diff
@@ -96,7 +96,7 @@ impl<'a, 'tcx> BlanketImplFinder<'a, 'tcx> {
                     name: None,
                     attrs: Default::default(),
                     visibility: Inherited,
-                    def_id: ItemId::Blanket { trait_: trait_def_id, for_: item_def_id },
+                    def_id: ItemId::Blanket { impl_id: impl_def_id, for_: item_def_id },
```

Comment:
Sorry, I was unclear: you don't need `for_` at all now, you can reconstruct it from the impl_id alone with `tcx.impl_trait_ref(impl_def_id).unwrap().self_ty()`.
```suggestion
                    def_id: ItemId::Blanket { impl_def_id, },
```

Reply:
Yes, I do need it, because apparently, only the implementation id is not enough to be unique (?????????).
When I only had the `impl_id`, I got a panic when documenting std that `assert_eq` on these two failed:
```
  left: `Item { id: Id("b:0:2454"), crate_id: 0, name: None, span: Some(Span { filename: "library/core/src/borrow.rs", begin: (208, 0), end: (213, 1) }), visibility: Default, docs: None, links: {}, attrs: [], deprecation: None, inner: Impl(Impl { is_unsafe: false, generics: Generics { params: [GenericParamDef { name: "T", kind: Type { bounds: [], default: None } }], where_predicates: [BoundPredicate { ty: Generic("T"), bounds: [TraitBound { trait_: ResolvedPath { name: "Sized", id: Id("0:2854"), args: Some(AngleBracketed { args: [], bindings: [] }), param_names: [] }, generic_params: [], modifier: Maybe }] }] }, provided_trait_methods: [], trait_: Some(ResolvedPath { name: "Borrow", id: Id("0:2448"), args: Some(AngleBracketed { args: [Type(Generic("T"))], bindings: [] }), param_names: [] }), for_: ResolvedPath { name: "Wrapping", id: Id("0:24138"), args: Some(AngleBracketed { args: [Type(Generic("T"))], bindings: [] }), param_names: [] }, items: [Id("0:2456")], negative: false, synthetic: false, blanket_impl: Some(Generic("T")) }) }`,
 right: `Item { id: Id("b:0:2454"), crate_id: 0, name: None, span: Some(Span { filename: "library/core/src/borrow.rs", begin: (208, 0), end: (213, 1) }), visibility: Default, docs: None, links: {}, attrs: [], deprecation: None, inner: Impl(Impl { is_unsafe: false, generics: Generics { params: [GenericParamDef { name: "T", kind: Type { bounds: [], default: None } }], where_predicates: [BoundPredicate { ty: Generic("T"), bounds: [TraitBound { trait_: ResolvedPath { name: "Sized", id: Id("0:2854"), args: Some(AngleBracketed { args: [], bindings: [] }), param_names: [] }, generic_params: [], modifier: Maybe }] }] }, provided_trait_methods: [], trait_: Some(ResolvedPath { name: "Borrow", id: Id("0:2448"), args: Some(AngleBracketed { args: [Type(Generic("T"))], bindings: [] }), param_names: [] }), for_: ResolvedPath { name: "ParseFloatError", id: Id("0:23084"), args: Some(AngleBracketed { args: [], bindings: [] }), param_names: [] }, items: [Id("0:2456")], negative: false, synthetic: false, blanket_impl: Some(Generic("T")) }) }`', src/librustdoc/json/mod.rs:179:17

```

Reply:
... @CraftSpider why are those separate? I would only expect there to be one item per blanket impl, generating a separate item for every type it *could* apply to seems both inefficient and not particularly useful.

Reply:
I have no clue, I'd expect the same. I'll look into the code tomorrow.

Reply:
Should we wait with this PR until you found a solution? We can also merge this now and make a follow up PR.

Reply:
Hmm, yeah this seems fine in the meantime. r=me once you fix the conflict.

Reply:
Agree with Josh, this isn't a blocking concern.

Reply:
Alright. @CraftSpider will you open the follow-up PR with the fix for not emitting duplicate blanket implementations and the fixed `ItemId`?